nservicebus/handlers/async-handlers.md

```diff
@@ -0,0 +1,163 @@
+---
+title: Asynchronous Handlers
+summary: Shows how to deal with synchronous and asynchronous code inside asynchronous handlers
+reviewed: 2016-07-05
+tags:
+ - Async
+---
+
+
+## Introduction
+
+[Message handlers](/nservicebus/handlers/) and [Sagas](/nservicebus/sagas/) will be invoked from an IO thread pool thread. Typically message handlers and sagas issue IO bound work like sending or publishing messages, storing information into databases, calling web services and more. In other cases, message handlers are used to schedule compute-bound work. To be able to write efficient message handlers and sagas it is crucial to understand the difference between compute-bound and IO bound work and the thread pools involved.
+
+
+### Threadpools
+
+There are two thread pools. The worker thread pool and the IO thread pool.
```

Comment:
There are two thread pools in the .NET Framework?
In Windows?

Can we add a little more context to this section?